mlir/lib/Dialect/SparseTensor/Transforms/Sparsification.cpp

```diff
@@ -1032,10 +1031,14 @@ static bool getAllTidLvlsInLatPoints(
       });
 
   if (isDenseLT(env.lt(outTid, curr))) {
+    auto stt = getSparseTensorType(env.op().getOutputs().front());
     // Note that we generate dense indices of the output tensor
     // unconditionally, since they may not appear in the lattice, but may be
     // needed for linearized env.
-    callback(env.makeTensorLevel(outTid, *outLvl), nullptr);
+    // TODO: we should avoid introducing corner cases for all-dense sparse
```

Comment:
it still feels like the all-dense case should not have to be different from any other combination of dense/compressed, so I agree with the TODO. Perhaps our lattices need to reflect that better.